crates/sui/src/upgrade_compatibility.rs

```diff
@@ -1371,11 +1615,150 @@ fn enum_variant_missing_diag(
     Ok(diags)
 }
 
+/// Return a diagnostic for a type parameter mismatch
+fn enum_type_param_mismatch(
+    enum_name: &Identifier,
+    old_enum: &Enum,
+    new_enum: &Enum,
+    compiled_unit_with_source: &CompiledUnitWithSource,
+    lookup: &IdentifierTableLookup,
+) -> Result<Diagnostics, Error> {
+    let mut diags = Diagnostics::new();
+
+    let enum_index = lookup
+        .enum_identifier_to_index
+        .get(enum_name)
+        .context("Unable to get enum index")?;
+
+    let enum_sourcemap = compiled_unit_with_source
+        .unit
+        .source_map
+        .get_enum_source_map(EnumDefinitionIndex::new(*enum_index))
+        .context("Unable to get enum source map")?;
+
+    let def_loc = enum_sourcemap.definition_location;
+
+    if old_enum.type_parameters.len() != new_enum.type_parameters.len() {
```

Comment:
I wonder if the logic below here could be abstracted out and shared between the enum and struct versions of this error?

Reply:
appears so, fixed